src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -934,7 +934,7 @@ public static unsafe void Scale(float scale, Span<float> dst)
                     result = Avx.And(result, leadingMask);
 
                     temp = Avx.Multiply(scaleVector256, temp);
-                    temp = Avx.Or(temp, result);
+                    temp = Avx.Add(temp, result);
```

Comment:
Instead of using `Add`, which is a little confusing to a new reader, would it make more sense to do the "trailing mask" after the `Avx.Multiply`? That way you can still use `Avx.Or`, which is the more reasonable/obvious thing to do here.

No matter which way we end up, I believe a comment is justified here stating why we are doing what we are doing.